src/ophys_etl/modules/segmentation/postprocess_utils/roi_merging.py

```diff
@@ -0,0 +1,1137 @@
+
+    min_chunk: int
+        Minimum acceptable chunk_size
+
+    denom_factor: int
+        number of chunks that should ultimately be
+        sent to each core (default=4)
+
+    Returns
+    -------
+    chunk_size: int
+    """
+    chunk_size = n_elements//(denom_factor*n_cores-1)
+    if chunk_size < min_chunk:
+        chunk_size = min_chunk
+    return chunk_size
+
+
+def _find_merger_candidates(roi_pair_list: List[Tuple[OphysROI, OphysROI]],
+                            dpix: float,
+                            output_list: multiprocessing.managers.ListProxy) -> None:
+    """
+    Find all of the abutting ROIs in a list of OphysROIs
+
+    Parameters
+    ----------
+    roi_pair_list: List[Tuple[OphysROI, OphysROI]]
+        A list of tuples of OphysROIs that need to be tested
+        to see if they abut
+
+    dpix: float
+       The maximum distance from each other two ROIs can be at
+       their nearest point and still be considered to abut
+
+    output_list: multiprocessing.managers.ListProxy
+        List where tuples of IDs of abutting ROIs will be written
+
+    Returns
+    -------
+    None
+        Appends tuples of (roi_id_0, roi_id_1) corresponding
+        to abutting ROIs into output_list
+    """
+    local = []
+    for pair in roi_pair_list:
+        if do_rois_abut(pair[0], pair[1], dpix=dpix):
+            local.append((pair[0].roi_id, pair[1].roi_id))
+    for pair in local:
+        output_list.append(pair)
+    return None
+
+
+def find_merger_candidates(roi_list: List[OphysROI],
+                           dpix: float,
+                           rois_to_ignore: Optional[set]=None,
+                           n_processors: int = 8) -> List[Tuple[int, int]]:
+    """
+    Find all the pairs of abutting ROIs in a list of OphysROIs.
+    Return a list of tuples like (roi_id_0, roi_id_1) specifying
+    the ROIs that abut.
+
+    Parameters
+    ----------
+    roi_list: List[OphysROI]
+
+    dpix: float
+       The maximum distance from each other two ROIs can be at
+       their nearest point and still be considered to abut
+
+    rois_to_ignore: Optional[set]
+       Optional set of ints specifying roi_id of ROIs not to consider
+       when looking for pairs. Note: a pair will only be ignored
+       if *both* ROIs are in rois_to_ignore. If one of them is not,
+       the pair is valid (default: None)
+
+    n_processors: int
+       Number of cores to use (this method uses multiprocessing since, for
+       full fields of view, there can be tens of millions of pairs of
+       ROIs to consider)
+
+   Returns
+   -------
+   output: List[Tuple[int, int]]
+       List of tuples of roi_ids specifying pairs of abutting ROIs
+    """
+    mgr = multiprocessing.Manager()
+    output_list = mgr.list()
+
+    n_rois = len(roi_list)
+
+    process_list = []
+
+    n_pairs = n_rois*(n_rois-1)//2
+    d_pairs = chunk_size_from_processors(n_pairs, n_processors, 100)
+
+    subset = []
+    for i0 in range(n_rois):
```

Comment:
double loop would be a lot shorter and cleaner looking using `itertools.combinations`
https://docs.python.org/3/library/itertools.html

Reply:
Thanks for pointing that out. Done.